test/appium/tests/critical/chats/test_public_chat_browsing.py

```diff
@@ -1245,3 +1244,199 @@ def test_community_join_when_node_owner_offline(self):
+
+        self.community_1, self.community_2 = self.home_1.get_community_view(), self.home_2.get_community_view()
+        self.community_1.invite_to_community(self.community_name, self.username_2)
+        self.channel_1 = self.home_1.get_to_community_channel_from_home(self.community_name)
+
+        self.community_2.join_community()
+        self.channel_2 = self.community_2.get_channel(self.channel_name).click()
+
+    @marks.testrail_id(741924)
+    def test_community_mobile_data_and_wi_fi_only_fetching(self):
+        self.device_2.just_fyi("Receiver turns wi-fi off")
+        self.device_2.navigate_back_to_home_view()
+        self.device_2.driver.set_network_connection(ConnectionType.DATA_ONLY)
+        # ToDo: pop up is not shown on LT emulators, needs more investigation
+        # self.device_2.turn_wi_fi_off()
+        #
+        # for translation in ['which-connection-to-use', 'mobile-and-wifi', 'wifi-only']:
+        #     if not self.device_2.element_by_translation_id(translation).is_element_displayed():
+        #         self.errors.append(self.device_2,
+        #                            "Element with text '%s' is not shown when switching to mobile data" % translation)
+        profile_2 = self.device_2.profile_button.click()
+        profile_2.syncing_button.scroll_and_click()
+        profile_2.sync_and_backup_button.click()
+        profile_2.wi_fi_only_button.click()
+        self.device_2.click_system_back_button(times=2)
+        self.home_2.communities_tab.click()
+        self.home_2.discover_communities_button.click()
+        if self.home_2.community_card_item.is_element_displayed(10):
+            self.errors.append(self.home_2,
+                               "Community can be fetched while using mobile data with wi-fi only syncing settings")
+
+        message_1 = 'message text 1'
+        self.channel_1.just_fyi("Sender sends a message in the community channel when receiver is using mobile data")
+        self.channel_1.send_message(message_1)
+        self.home_2.click_system_back_button()
+        self.home_2.get_to_community_channel_from_home(self.community_name)
+        if not self.channel_2.chat_element_by_text(message_1).is_element_displayed(120):
+            self.errors.append(self.device_2,
+                               "Can't receive the message '%s' in community when using mobile data" % message_1)
+
+        self.device_2.just_fyi("Receiver turns internet off")
+        self.device_2.driver.set_network_connection(ConnectionType.NO_CONNECTION)
+        message_2 = 'message text 2'
+        self.channel_1.just_fyi("Sender sends messages in the community channel and 1-1 chat when receiver is offline")
+        self.channel_1.send_message(message_2)
+        self.device_1.navigate_back_to_home_view()
+        self.device_1.chats_tab.click()
+        self.home_1.get_chat(self.username_2).click()
+        message_3 = 'message text 3'
+        self.chat_1.send_message(message_3)
```

Comment:
better to add here a check that it is SENT already (status is sent), as if it will be sent when DATA_ONLY is set, the test will be flaky.